src/main/telemetry/srxl.c

```diff
@@ -155,17 +159,54 @@ typedef struct
 
 #define SPEKTRUM_RPM_UNUSED 0xffff
 #define SPEKTRUM_TEMP_UNUSED 0x7fff
+#define MICROSEC_PER_MINUTE 60000000
 
-bool srxlFrameRpm(sbuf_t *dst, timeUs_t currentTimeUs)
+//Original range of 1 - 65534 uSec gives an RPM range of 915 - 60000000rpm, 60MegaRPM
+#define SPEKTRUM_MIN_RPM 999      // Min RPM to show the usre, indicating RPM is really below 999
+#define SPEKTRUM_MAX_RPM 60000000
+
+uint16_t getMotorAveragePeriod(void)
 {
+
+#if defined( USE_ESC_SENSOR_TELEMETRY) || defined( USE_DSHOT_TELEMETRY)
+    uint32_t rpm = 0;
     uint16_t period_us = SPEKTRUM_RPM_UNUSED;
-#ifdef USE_ESC_SENSOR_TELEMETRY
+
+#if defined( USE_ESC_SENSOR_TELEMETRY)
     escSensorData_t *escData = getEscSensorData(ESC_SENSOR_COMBINED);
-    if (escData != NULL && escData->rpm > 0) {
-        period_us = 60000000 / escData->rpm; // revs/minute -> microSeconds
+    if (escData != NULL) {
+        rpm = escData->rpm;
+    }
+#endif
+
+#if defined( USE_DSHOT_TELEMETRY)
+    if (useDshotTelemetry) {
+        uint16_t motors = getMotorCount();
+
+        if (motors > 0) {
+            for (int  motor = 0; motor < motors; motor++) {
+                rpm += getDshotTelemetry(motor);
+            }
+            rpm = 100.0f / (motorConfig()->motorPoleCount / 2.0f) * rpm;  // convert erpm freq to RPM.
```

Comment:
One could avoid costly division here by precomputing the reciprocal.

Reply:
OK, example please. This is C&P from osd.c....

Reply:
This is @joelucid is doing it in `rpm_filter.c`: https://github.com/betaflight/betaflight/pull/7757#discussion_r264055925

But I think this is fine for now, the a getter that directly returns RPM should be added to the bidirectional Dshot at a later stage.